editors/code/src/command/workspace-folder-formatting.ts

```diff
@@ -0,0 +1,228 @@
+ * Because we go through the Air CLI here, forward propagated document settings
+ * are ignored, so even if you don't have an `air.toml`, you will get Air's
+ * default indentation settings.
+ *
+ * We can't simply require an `air.toml` at the workspace folder root to avoid
+ * confusion here, because you may instead have an `air.toml` at
+ * `{root}/R/air.toml` that would handle any R files in your project. Instead,
+ * we hope this isn't common enough to come up much in practice.
+ */
+export function workspaceFolderFormatting(ctx: Ctx): Cmd {
+	return async () => {
+		const binaryPath = ctx.lsp.getBinaryPath();
+
+		const workspaceFolder = await selectWorkspaceFolder();
+		if (!workspaceFolder) {
+			return;
+		}
+
+		const allTabsClosed = await closeAllTabs(workspaceFolder);
+		if (!allTabsClosed) {
+			return;
+		}
+
+		const workspaceFolderPath = workspaceFolder.uri.fsPath;
+
+		let stderr = "";
+		let anyErrors = false;
+
+		// i.e., `air format {workspaceFolderPath} --no-color`
+		const args = ["format", workspaceFolderPath, "--no-color"];
+
+		// This should not matter since the path is explicitly supplied, but better to be safe
+		const options = {
+			cwd: workspaceFolderPath,
+		};
+
+		// A promise that resolves when the spawned process closes or errors
+		const finishedFormatting = new Promise<void>((resolve) => {
+			// Use spawn instead of exec to avoid maxBufferExceeded error
+			const p = cp.spawn(binaryPath, args, options);
+
+			p.stderr.setEncoding("utf8");
+			p.stderr.on("data", (data) => (stderr += data));
+
+			p.on("error", () => {
+				anyErrors = true;
+				return resolve();
+			});
+
+			p.on("close", (code) => {
+				if (code !== 0) {
+					anyErrors = true;
+				}
+				return resolve();
+			});
+		});
+
+		await finishedFormatting;
+
+		if (anyErrors) {
+			output.log(
+				`Errors occurred while formatting the ${workspaceFolder.name} workspace folder.\n${stderr}`,
+			);
+
+			const answer = await vscode.window.showInformationMessage(
+				`Errors occurred while formatting the ${workspaceFolder.name} workspace folder. View the logs?`,
+				{ modal: true },
+				"Yes",
+				"No",
+			);
+
+			if (answer === "Yes") {
+				output.show();
+			}
+
+			return;
+		}
+
+		vscode.window.showInformationMessage(
+			`Successfully formatted the ${workspaceFolder.name} workspace folder.`,
+		);
+	};
+}
+
+async function selectWorkspaceFolder(): Promise<
+	vscode.WorkspaceFolder | undefined
+> {
+	const workspaceFolders = vscode.workspace.workspaceFolders;
+
+	if (!workspaceFolders || workspaceFolders.length === 0) {
+		vscode.window.showErrorMessage(
+			"You must be inside a workspace to format a workspace folder.",
+		);
+		return undefined;
+	}
+
+	if (workspaceFolders.length === 1) {
+		return workspaceFolders[0];
+	}
+
+	// Let the user select a workspace folder if >1 are open, may be
+	// `undefined` if user bails from quick pick!
+	const workspaceFolder =
+		await selectWorkspaceFolderFromQuickPick(workspaceFolders);
```

Comment:
In the common case of just 1 workspace folder, we don't even ask the user which to use